storage/rocksdb/rdb_buff.h

```diff
@@ -359,6 +359,21 @@ class Rdb_string_writer {
     DBUG_ASSERT(pos < get_current_pos() && (pos + 1) < get_current_pos());
     rdb_netbuf_store_uint16(m_data.data() + pos, new_val);
   }
+
+  void truncate(const size_t &pos) {
+    DBUG_ASSERT(0 <= pos && pos < m_data.size());
+    m_data.resize(pos);
+  }
+
+  void allocate(const uchar &val, const size_t &len) {
```

Comment:
Probably better to swap these arguments, and have val be 0 by default, so we can just call allocate(len).